src/components/Microsoft.DotNet.UpgradeAssistant/TargetFramework/TargetFrameworkSelector.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.DotNet.UpgradeAssistant.TargetFramework
+{
+    public class TargetFrameworkSelector : ITargetFrameworkSelector
+    {
+        private const string DefaultCurrentTFMBase = "net5.0";
+        private const string DefaultLTSTFMBase = "net5.0";
+
+        private readonly ITargetFrameworkMonikerComparer _comparer;
+        private readonly IEnumerable<ITargetFrameworkSelectorFilter> _selectors;
+
+        private readonly string _currentTFMBase;
+        private readonly string _ltsTFMBase;
+
+        private readonly UpgradeTarget _upgradeTarget;
+
+        public TargetFrameworkSelector(UpgradeOptions options, ITargetFrameworkMonikerComparer comparer, IOptions<TFMSelectorOptions> selectorOptions, IEnumerable<ITargetFrameworkSelectorFilter> selectors)
+        {
+            _comparer = comparer;
+            _selectors = selectors;
+
+            _currentTFMBase = selectorOptions?.Value.CurrentTFMBase ?? DefaultCurrentTFMBase;
+            _ltsTFMBase = selectorOptions?.Value.LTSTFMBase ?? DefaultLTSTFMBase;
+            _upgradeTarget = options?.UpgradeTarget ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public async ValueTask<TargetFrameworkMoniker> SelectTargetFrameworkAsync(IProject project, CancellationToken token)
```

Comment:
Not directly tied to this change, but why do we call this method twice?

We invoke from `SetTFMStep.cs` during Initialize and during Apply

Reply:
In initialize to see if we have set it to something that we expect, and the apply to to get what to set it to. We could cache it for later, but apparently don't